src/node.cc

```diff
@@ -967,19 +967,21 @@ Local<Value> UVException(Isolate* isolate,
 // Look up environment variable unless running as setuid root.
 bool SafeGetenv(const char* key, std::string* text) {
 #ifndef _WIN32
-  if (getuid() != geteuid() || getgid() != getegid()) {
-    text->clear();
-    return false;
-  }
+  if (getuid() != geteuid() || getgid() != getegid())
+    goto fail;
 #endif
-  if (linux_at_secure) {
-    text->clear();
-    return false;
-  }
+
+#if defined(__linux__)
```

Comment:
Hm, we could skip the `#ifdef __linux__` ugliness mentioned there, and just add `linux_at_secure` into the existing `if (getuid() != geteuid() || getgid() != getegid())` check. Whichever you prefer.